### ISSUE TITLE
Queues with different ApproximateNumberOfMessages not equal

### ISSUE THREAD

**Issue by oman36** [spulec/moto#2386]
**Expected Behavior:**
Assume we call create_queue method of SQS resource with some queue attributes that are not controlled by create_queue method. If they contradict with existing queue attributes then QueueNameExists is raised

**Actual Behavior:**
Error is raised if ApproximateNumberOfMessages are different.

**Steps to Reproduce:**
```python
In [1]: from moto import mock_sqs
In [2]: mock_sqs().start()
In [3]: import boto3
In [4]: sqs = boto3.resource('sqs', region_name='us-east-1')
In [5]: queue = sqs.create_queue(QueueName="test-q-1")
In [6]: queue.send_message(MessageBody='test')
In [7]: queue = sqs.create_queue(QueueName="test-q-1")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/user/project/venv/lib/python3.7/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/home/user/project/venv/lib/python3.7/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/home/user/project/venv/lib/python3.7/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/user/project/venv/lib/python3.7/site-packages/botocore/client.py", line 661, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.QueueNameExists: An error occurred (QueueAlreadyExists) when calling the CreateQueue operation: The specified queue already exists.
```

**Comment by oman36** [comment]
#2387

**Comment by spulec** [comment]
Closed with #2387
